src/Form/DomainMenuAccessSettingsForm.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+namespace Drupal\domain_menu_access\Form;
+
+use Drupal\Component\Plugin\Discovery\CachedDiscoveryInterface;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Entity\EntityStorageInterface;
+use Drupal\Core\Form\ConfigFormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+class DomainMenuAccessSettingsForm extends ConfigFormBase {
+
+  /**
+   * Block manager
+   */
+  protected $blockManager;
+
+  /**
+   * Menu entities storage
+   */
+  protected $menu;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(ConfigFactoryInterface $config_factory, CachedDiscoveryInterface $block_manager, EntityStorageInterface $menu_storage) {
+    parent::__construct($config_factory);
+    $this->blockManager = $block_manager;
+    $this->menu = $menu_storage;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('config.factory'),
+      $container->get('plugin.manager.block'),
+      $container->get('entity.manager')->getStorage('menu')
```

Comment:
great, but should be `$container->get('entity_type.manager')->getStorage('menu')`